barman/backup.py

```diff
@@ -370,6 +370,24 @@ def get_last_backup_id(self, status_filter=DEFAULT_STATUS_FILTER):
         ids = sorted(available_backups.keys())
         return ids[-1]
 
+    def get_last_full_backup_id(self, status_filter=DEFAULT_STATUS_FILTER):
+        """
+        Get the id of the latest/last FULL backup in the catalog (if exists)
+
+        :param status_filter: The status of the backup to return,
+            default to DEFAULT_STATUS_FILTER.
+        :return string|None: ID of the backup
+        """
+        available_backups = self.get_available_backups(status_filter)
```

Comment:
I wonder if we should just filter out whatever has not parent = none.
something like:
```python
    available_full_backups = sorted(
        list(filter(lambda x: x.parent_backup_id is None,
                    self.get_available_backups(status_filter))))
    if len(available_full_backups) == 0:
        return None

    ids = sorted(available_full_backups.keys())
    return ids[-1]
```
I thought about this because I expect the number of incremental backups for a server to be higher compared to full backups... and is totally possible that we have to discard a non-small amount of incrementals before finding the first full.
The above implementation is technically quicker as we use a primitive chain of calls that has an underlying implementation in C..
But is possibly an overkill and maybe less readable.
@barthisrael @andremagui @bleidens and @gustabowill  thoughts on this?

Reply:
We don't know how people will play with this in practice, but I guess your code is clean enough to read.

I'd only say to create a couple of methods in `LocalBackupInfo` maybe, let's say:

* `is_full_backup`: check if mode is postgres, summarize is on, parent is None
* `is_incremental_backup`: check if mode is postgres, summarize is on, parent is not None

Or some other name? Would worth it a note in the docstring of the method, something like:

```python
"""
    (...)
    .. note::
        Only consider backups which are eligible for Postgres core incremental backups
    (...)
"""
```

Then we could have something like this:

```python
list(filter(lambda x: x.is_full_backup(),
```

Any thoughts?

Reply:
LTGM. It looks cleaner having this split in methods and applied like that! I do agree with @gcalacoci implementation and @barthisrael solution so we have something more readable.

Reply:
I'm not sure that `is_full_backup` as the name of the method is "correct" as theoretically a pg_basebackup executed backup that has the summarize on, is full... but is not eligible for incremental... 
OTOH `is_full_and_incremental_eligible` is a terrible name 😓

Reply:
has_parent_backup_id?

Reply:
> I'm not sure that `is_full_backup` as the name of the method is "correct" as theoretically a pg_basebackup executed backup that has the summarize on, is full... but is not eligible for incremental... OTOH `is_full_and_incremental_eligible` is a terrible name 😓 @barthisrael @andremagui

Right, that's why I thought of using the docstring note, which will show on hover in the IDE e.g.

I don't have many ideas of better names 😆

Reply:
formally is the most correct, but i cannot stand it